new yaml-config throws permission error for gcloud storage bucket -> old config is working

`Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (ftl.cli.firebase.test.android.AndroidRunCommand@70efb718): com.google.cloud.storage.StorageException: xxx@yyy.com does not have storage.buckets.list access to project 1234.`

`Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden
{
  "code" : 403,
  "errors" : [ {
    "domain" : "global",
    "message" : "xxx@yyy.com does not have storage.buckets.list access to project 1234.",
    "reason" : "forbidden"
  } ],
  "message" : "xxx@yyy.com does not have storage.buckets.list access to project 1234."
}`

***old yaml config***
```yaml
gcloud:
  results-bucket: 123
  results-dir: build-$BITRISE_BUILD_NUMBER
  record-video: true
  timeout: 30m
  async: false
  project: 1234
  results-history-name: xxx
  flaky-test-attempts: 1
  app: $BITRISE_APK_PATH
  test: $BITRISE_TEST_APK_PATH
  auto-google-login: false
  use-orchestrator: true
  environment-variables:
    clearPackageData: true
  performance-metrics: true
  device:
  - model: walleye
    version: 28
    locale: en
    orientation: portrait

flank:
  testShards: 8
  shardTime: -1
  repeatTests: 1
  smartFlankGcsPath: gs://xxx/yyy/results.xml
  disableSharding: false
```

***new yaml config***
```yaml
gcloud:
  results-bucket: 123
  results-dir: build-$BITRISE_BUILD_NUMBER
  record-video: true
  timeout: 30m
  async: false
  results-history-name: xxx
  flaky-test-attempts: 1
  app: $BITRISE_APK_PATH
  test: $BITRISE_TEST_APK_PATH
  auto-google-login: false
  use-orchestrator: true
  environment-variables:
    clearPackageData: true
  performance-metrics: true
  device:
  - model: walleye
    version: 28
    locale: en
    orientation: portrait

flank:
  max-test-shards: 8
  shard-time: -1
  repeat-tests: 1
  smart-flank-gcs-path: gs://xxx/yyy/results.xml
  disable-sharding: false
  project: 1234
```

--- a/test_runner/gradle/wrapper/gradle-wrapper.properties
+++ b/test_runner/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.3-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/test_runner/src/main/kotlin/ftl/args/ArgsHelper.kt
+++ b/test_runner/src/main/kotlin/ftl/args/ArgsHelper.kt
@@ -24,7 +24,6 @@ import ftl.shard.Shard
 import ftl.shard.StringShards
 import ftl.shard.stringShards
 import ftl.util.Utils
-import ftl.util.Utils.fatalError
 import java.io.File
 import java.net.URI
 import java.nio.file.Files
@@ -103,11 +102,14 @@ object ArgsHelper {
     fun createJunitBucket(projectId: String, junitGcsPath: String) {
         if (FtlConstants.useMock || junitGcsPath.isEmpty()) return
         val bucket = junitGcsPath.drop(GCS_PREFIX.length).substringBefore('/')
-            createGcsBucket(projectId, bucket)
+        createGcsBucket(projectId, bucket)
     }
 
+    // Make best effort to list/create the bucket.
+    // Due to permission issues, the user may not be able to list or create buckets.
     fun createGcsBucket(projectId: String, bucket: String): String {
-        if (bucket.isEmpty()) return GcToolResults.getDefaultBucket(projectId) ?: throw RuntimeException("Failed to make bucket for $projectId")
+        if (bucket.isEmpty()) return GcToolResults.getDefaultBucket(projectId)
+            ?: throw RuntimeException("Failed to make bucket for $projectId")
         if (useMock) return bucket
 
         // test lab supports using a special free storage bucket
@@ -122,21 +124,29 @@ object ArgsHelper {
         val storageLocation = "us-central1"
 
         val bucketListOption = Storage.BucketListOption.prefix(bucket)
-        val storageList = storage.list(bucketListOption).values?.map { it.name } ?: emptyList()
+        val storageList = emptyList<String>()
+
+        try {
+            storage.list(bucketListOption).values?.map { it.name }
+        } catch (e: Exception) {
+            // User may not have list permission
+        }
+
         if (storageList.contains(bucket)) return bucket
 
         try {
-            return storage.create(
+            storage.create(
                 BucketInfo.newBuilder(bucket)
                     .setStorageClass(StorageClass.REGIONAL)
                     .setLocation(storageLocation)
                     .setLabels(bucketLabel)
                     .build()
-            ).name
-        } catch (e: com.google.cloud.storage.StorageException) {
-            fatalError("Failed to create bucket $bucket\n${e.message}")
-            throw RuntimeException("will not throw")
+            )
+        } catch (e: Exception) {
+            // User may not have create permission
         }
+
+        return bucket
     }
 
     private fun serviceAccountProjectId(): String? {

